xof.go

```diff
@@ -0,0 +1,53 @@
+package kyber
+
+import (
+	"crypto/cipher"
+)
+
+// Sponge is an interface representing a primitive sponge function.
+type Sponge interface {
```

Comment:
Hmmm, not sure if it still makes sense to keep this intermediate Sponge interface or the associated code.  Is it or will it be actually used in the SHAKE and/or BLAKE2 implementations of the XOF interface?